go/vt/vtgate/planbuilder/physical/subquery_planning.go

```diff
@@ -107,10 +138,17 @@ func mergeSubQueryOp(ctx *plancontext.PlanningContext, outer *Route, inner *Rout
 		outer.SysTableTableName[k] = v
 	}
 
-	err = outer.resetRoutingSelections(ctx)
-	if err != nil {
-		return nil, err
+	outer.Selected = nil
+	switch outer.RouteOpCode {
+	case engine.DBA, engine.Next, engine.Reference, engine.Unsharded:
```

Comment:
I've taken this list from `resetRoutingSelections`, but I'm not sure we can actually ever reach this code with one of these routing op codes? 🤔

Reply:
I am unsure too. I am assuming if, for instance, we had a DBA query with a subquery? cc @systay